tools/json_tools/gun_variant_validator.py

```diff
@@ -161,6 +161,7 @@
     "needlegun",
     "needlepistol",
     "rm298",
+    "ruger_pcc"
```

Comment:
Please don't add to this! The way to fix this is to make the name of the non-variant item contain `Glock` (e.g. `Glock 9mm pistol caliber carbine`).

Reply:
It's NOT a "Glock 9mm pistol calibre carbine" though, it's a Ruger, and it doesn't have a variant.
Is this why there are a bunch of items with single variants defined?

Reply:
My bad, I missed that. Yes, this is why there are many guns with single variants.

Reply:
Ok so essentially we're defining a "less technically accurate but more helpful" name for it to use when the "Detailed gun names" option is off.  Feels weird for that to be the "main" description and the more accurate variant to be the real name, but that's ok.

Reply:
here's what I'm thinking of doing. There are a number of these companion carbines cut from the same mold, and they can aggregate under this entry.
```
    "id": "glock_mag_pcc",
    "name": { "str": "9mm pistol caliber carbine (Glock magazines)" },
    "description": "This Pistol Caliber Carbine is a semi-automatic pistol-caliber carbine chambered in 9mm Parabellum, designed as a companion to certain semi-automatic pistols, using the same cartridges and magazines.",
    "variants": [
      {
        "id": "ruger_pcc",
        "name": { "str": "Ruger 9mm pistol caliber carbine" },
        "description": "The Ruger PC-9 Pistol Caliber Carbine is a semi-automatic pistol-caliber carbine chambered in 9mm Parabellum, designed as a companion to certain Ruger P series semi-automatic pistols, using the same cartridges and magazines of the P-Series pistols."
      }
    ],
```

Reply:
`Glock pistol caliber carbine` reads more like it describes a single gun than a category, which I think fits better for the non-detailed gun names treating the whole category as one gun. I did similar for the vector: https://github.com/CleverRaven/Cataclysm-DDA/blob/d30757ea5f583f025dead7df832a75c3d930f855/data/json/items/gun/45.json#L12-L21

It doesn't matter though, as long as Glock is in the name - I know people find the vector name weird (I've seen a number of comments on it).

I'm also assuming the "show caliber with gun name" option is turned on, which makes the `9mm` in the name redundant.

Reply:
Even if we're doing a simplified version sticking Glock in front bothers me because that very much says to me "this is a gun made by Glock" That would be fine if there WAS a Glock carbine that fits, but as far as I can tell there isn't.
Thanks for the note about the calibre, if they want the clarification they can have it.